cmd/kubeadm/app/phases/certs/certlist.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certs
+
+import (
+	"crypto/x509"
+
+	certutil "k8s.io/client-go/util/cert"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/certs/pkiutil"
+)
+
+type KubeadmCert struct {
+	Config        certutil.Config
+	Name          string
+	BaseName      string
+	AltNames      *certutil.AltNames
+	DaughterCerts []KubeadmCert
+}
+
+// GetKubeadmCerts returns a list of all certificates, given the passed configuration.
+func GetKubeadmCerts(cfg *kubeadmapi.MasterConfiguration) ([]KubeadmCert, error) {
```

Comment:
I would expect the return type to be `([]*KubeadmCert, error)` -- what's the reason behind returning the non-pointer struct?